account_payment_order/models/account_move.py

```diff
@@ -57,6 +57,16 @@ def _get_payment_order_communication_direct(self):
                 communication = self.ref or self.payment_reference
             else:
                 communication = self.payment_reference or self.name
+            default_payment_type = self._context.get("default_payment_type", False)
```

Comment:
Why are you using the default_payment_type?

If you want to use name, just leave payment_reference and ref to false :thinking: